pennylane/qchem/openfermion_obs.py

```diff
@@ -917,6 +892,120 @@ def molecular_hamiltonian(
     + (0.176276408043196) [Z2 Z3]
     """
 
+    if len(args) != 0:
+        return _molecular_hamiltonian_dispatch(*args, **kwargs)
+
+   method = kwargs.get("symbols", None) or kwargs.get("molecule", None)
+   
+   if method is not None:
+       kwargs.pop(method)
+       return _molecular_hamiltonian_dispatch(method, **kwargs)
```

Comment:
Forgot something, this should work.

```suggestion
    method = kwargs.get("symbols", None) or kwargs.get("molecule", None)

    if method is not None:
        kwargs.pop("molecule" if isinstance(method, Molecule) else "symbols", None)
        return _molecular_hamiltonian_dispatch(method, **kwargs)
```